sbctl.go

```diff
@@ -15,28 +16,78 @@ import (
 
 // Functions that doesn't fit anywhere else
 
-// Veryvery simple check
+type LsblkEntry struct {
+	Parttype   string `json:"parttype"`
+	Mountpoint string `json:"mountpoint"`
+	Pttype     string `json:"pttype"`
+	Fstype     string `json:"fstype"`
+}
+
+type LsblkRoot struct {
+	Blockdevices []LsblkEntry `json:"blockdevices"`
+}
+
+// Slightly more advanced check
 func GetESP() string {
-	if _, err := os.Stat("/efi"); !os.IsNotExist(err) {
-		return "/efi"
+
+	envEspPath, found := os.LookupEnv("SYSTEMD_ESP_PATH")
```

Comment:
`for _, env := range []string{"SYSTEMD_ESP_PATH", "ESP_PATH"} {`

Less code duplication for the sake of a for loop.